packages/jazz-tools/src/coValues/coList.ts

```diff
@@ -446,6 +449,77 @@ export class CoList<Item = any> extends Array<Item> implements CoValue {
     ): InstanceType<Cl> {
         return cl.fromRaw(this._raw) as InstanceType<Cl>;
     }
+
+    asPlainData<L extends CoList>(this: L): RecursiveCoMapInit<L>;
+    asPlainData<L extends CoList, Depth>(
+        this: L,
+        depth: Depth & DepthsIn<L>,
+    ): Promise<RecursiveCoMapInit<L> | undefined>;
+    asPlainData<L extends CoList, Depth>(
+        this: L,
+        depthOrSeen?:
+            | (Depth & DepthsIn<L>)
+            | WeakMap<CoMap | CoList, RecursiveCoMapInit<L>>,
+    ): RecursiveCoMapInit<L> | Promise<RecursiveCoMapInit<L> | undefined> {
+        if (depthOrSeen instanceof WeakMap) {
+            return this.asPlainDataSync(depthOrSeen);
+        } else if (depthOrSeen === undefined) {
+            return this.asPlainDataSync(new WeakMap());
```

Comment:
**question:** why using a `WeakMap` here instead of a `Map`?

`Map` should be a little faster than `WeakMap`, so if we don't have a particular reason to use `WeakMap` we should favor the usage of `Map`

Reply:
Since we only use the Map to collect seen instances and don't expose this to the outside after we return, I think we can indeed use just a normal `Map` and it will be GC'd just fine.